readalongs/align.py

```diff
@@ -33,46 +33,58 @@
 from readalongs.text.util import save_xml
 from readalongs.log import LOGGER
 
-####
-#
-# Some distros (Python2, Python3 on Windows it seems) don't have the WAV
-# reading methods being context managers; the following checks whether the
-# necessary methods are present and if not, add thems.
-#
-# Based on http://web.mit.edu/jgross/Public/21M.065/sound.py 9-24-2017
-####
-
 
-def _trivial__enter__(self):
-    return self
+def correct_adjustments(start: Union[int, float], end: Union[int, float], do_not_align_segments: List[object]) -> (int, int):
+    """ Given the start and end of a segment and a list of do-not-align segments,
+        If one of the do-not-align segments occurs inside one of the start-end range,
+        align the start or end with the do-not-align segment, whichever requires minimal change
+    """
+    if isinstance(start, float):
+        LOGGER.info(f'Converting {start} to milliseconds')
+        start = int(start * 1000)
+    if isinstance(end, float):
+        LOGGER.info(f'Converting {end} to milliseconds')
+        end = int(end * 1000)
```

Comment:
This seems dangerous to me. It means that 100 is time=.1s, but 100.0 is time 100s. That's going to be a serious gotcha! If we want to support both s and ms, I would rather have different variable names for them...

Reply:
Yeah, I'm not a big fan of this back and forth either. The problem is that the library we're using to manipulate audio (pydub) works in milliseconds and the SMIL files have all used seconds. I'm not sure if the SMIL can be changed, but I don't really anticipate this being a gotcha - so long as we know that internally we use milliseconds and in the SMIL we print to seconds we should be fine - there's no scenario where someone should be manually entering anything here. All that said, I would certainly be in favour of just doing everything in milliseconds if that would be possible.

Reply:
see 18b713d

Reply:
Yeah, that's much better. Calling it `*_ms` makes it nicely self-documenting, and adding the comments in the functions.
We can't remove having to deal with both, but this makes it clear.